File: ObjC/PlusMinus.m
//Given an array of integers, calculate the ratios of its elements that are positive, negative, and zero.

- (void) plusMinus:(NSArray *)arr {
    NSMutableArray *positiveArr = [[NSMutableArray alloc] init];
    NSMutableArray *negativeArr = [[NSMutableArray alloc] init];
    for (NSNumber *n in arr) {
        if ([n intValue] > 0) {
            [positiveArr addObject:n];
        }
        else if ([n intValue] < 0) {
            [negativeArr addObject:n];
        }
    }
    float arrCount = arr.count;
    float positiveRatio = positiveArr.count/arrCount;
    float negativeRatio = negativeArr.count/arrCount;
    float zeroRatio = (arrCount - positiveArr.count - negativeArr.count)/arrCount;
    printf("%.6f\n", positiveRatio);
    printf("%.6f\n", negativeRatio);
    printf("%.6f", zeroRatio);
}

File: Swift/PlusMinus.swift
//Given an array of integers, calculate the ratios of its elements that are positive, negative, and zero.

func plusMinus(arr: [Int]) -> Void {
    var positiveArr = [Int]()
    var negativeArr = [Int]()
    var zeroArr = [Int]()
    arr.map { element in
        if element < 0 { negativeArr.append(element) }
        else if element > 0 { positiveArr.append(element) }
        else { zeroArr.append(element) }
    }
    let total = arr.count
    print(Float(positiveArr.count) / Float(total))
    print(Float(negativeArr.count) / Float(total))
    print(Float(zeroArr.count) / Float(total))
}

let arr = [1, -2, -7, 9, 1, -8, -5]
plusMinus(arr: arr)

